plugins/jobs/src/js/JobDelete.js

```diff
@@ -0,0 +1,70 @@
+import * as React from "react";
+import { componentFromStream } from "data-service";
+import { deleteJob } from "#SRC/js/events/MetronomeClient";
+import { Subject } from "rxjs/Subject";
+import "rxjs/add/operator/map";
+import "rxjs/add/operator/switchMap";
+import "rxjs/add/operator/do";
+import "rxjs/add/operator/combineLatest";
+import "rxjs/add/operator/startWith";
+import "rxjs/add/operator/catch";
+
+import JobDeleteModal from "./components/JobDeleteModal";
+
+function errorIsTaskCurrentRunning(errorMsg) {
+  return /stopCurrentJobRuns=true/.test(errorMsg);
+}
+
+function executeDelete({ jobId, stopCurrentJobRuns, onSuccess, errorMsg }) {
+  return deleteJob(jobId, stopCurrentJobRuns)
+    .map(_ => ({ done: true, stopCurrentJobRuns, errorMsg }))
+    .do(_ => onSuccess())
```

Comment:
🤔 wonder if `.do(onSuccess)` would do

Reply:
It will change the semantic and pass the value emited from the stream to onSucess. I don't want it to happen.